### ISSUE TITLE
pure virtual method called

### ISSUE THREAD

**Issue by marklit** [duckdb/duckdb_spatial#277]
This is running on Arch Linux on my Steam Deck. It has 16 GB of RAM and there is ~200 GB of free space atm.

```bash
$ aws s3 --no-sign-request cp s3://naturalearth/10m_cultural/ne_10m_admin_0_countries.zip ./
$ unzip ne_10m_admin_0_countries.zip
$ pip install duckdb h3 rich # DuckDB 0.10
$ wc -l h3_5s.txt # 14,290 records
$ python3 # 3.11
```

```python
import duckdb
import h3
from rich.progress import track


def get_iso3s():
    lookups = {x: h3.h3_to_geo(x)
               for x in open('h3_5s.txt').read().strip().splitlines() 
               if len(x) > 14}

    con = duckdb.connect(database=":memory:")
    con.execute('INSTALL spatial')
    con.execute('LOAD spatial')

    out = {}

    sql_iso = """select ISO_A3 
                 from st_read('ne_10m_admin_0_countries.shx') 
                 where st_contains(geom, st_point(?, ?)) 
                 limit 1;"""

    for h3_5, (lon, lat) in track(lookups.items()):
        iso3 = con.sql(sql_iso, params=(lon, lat)).fetchall()

        if iso3:
            out[h3_5] = iso3[0][0]

    return out


iso3s = get_iso3s()
```

I get the following returned around ~80% of the way through the ~14K lookups.

```
pure virtual method called
terminate called without an active exception
Aborted (core dumped)
```

**Comment by marklit** [comment]
Attempt number 20 or so managed to finish without issue. I'm not sure if there is something wrong with my system or the OS.

**Comment by Maxxen** [comment]
Hi!
It sounds like there's a bunch of different issues at play here. The pure virtual abort in particular seems like it would be hard to reproduce and could originate either in spatial, the python bindings or gdal. Regarding the GPKG warning/error it could just be that gdals arrow api can't handle non-conformant gpkgs. I'll try to have a look after next release.

Spatial has also been shipping an undocumented prototype native shapfile reader function, `ST_ReadSHP`. It doesn't handle all kinds of text encoding options and it needs a lot more testing but it could maybe act as a temporary workaround for your issue.

**Comment by marklit** [comment]
If you have any CLI commands I could run to try and pinpoint where the problem is I can run them on my Steam Deck.

I'll have a look into that new ST command.

**Comment by rouault** [comment]
> Error: IO Error: GDAL Error (1): too many arguments on function OGR_GPKG_FillArrowArray_INTERNAL

This is https://github.com/OSGeo/gdal/issues/8757 , which was fixed in GDAL 3.8.1

**Comment by Maxxen** [comment]
I don't think this issue is very actionable and a ton has changed since it was first reported. Please reopen if the issue still persists.
